src/libcore/hash/sip.rs

```diff
@@ -193,6 +193,14 @@ impl Hasher for SipHasher {
     }
 
     #[inline]
+    fn delimit(&mut self, len: usize) {
+        // skip the first delimiter
+        if self.length > 0 {
+            self.write_usize(len);
```

Comment:
I can make more than two, although it's limited to no more than N + 1 interpretations for an N-byte string.  This example achieves `N / usize::BYTES`:

```rust
#[derive(Hash)]
enum Foo { One, Two(Box<Foo>) }

(vec![2usize; n], One)
```

(My experiments with `--pretty expanded` tell me that discriminators are one-based, not zero-based).